docs/source/user_guide/12_list_of_environment_variables.rst

```diff
@@ -0,0 +1,149 @@
+Referring Env variable in your code for hyperlink (Usage Example)
+------------------------------------------------------------------
+
+Your documentation text while using/referring env like
+this ``:envvar:`RADICAL_TEST_ENV``` and continuing.
+
+List of Env vars
+__________________
+
+
+.. list-table:: User List
+    :widths: 35 100 20
+    :header-rows: 1
+
+    * - Environment Variable Name
+      - Description
+      - Default value
+    * - .. envvar:: RADICAL_BASE
```

Comment:
```suggestion
    * - :envvar:`RADICAL_BASE`
```
Since this is not the place that you are providing a canonical documentation object, you should use the *role*, not the *directive*. It will format the way you want and will automatically link once you add a directive in the separate doc you describe.

If I have misunderstood you, and you intend for *this* to be the document that provides the definition text for the environment variable, then I think it hurts the documentation to (mis)use the directives in this way.

1. It is useful to let the directive tell sphinx what the definition is so that it can provide more than just a link.
2. Are you sure that each environment variable will fall neatly into one of these lists exactly once?

If the formatting of the environment variable definition (or of the text around any directive) is really important, you should probably be customizing a style sheet (or writing a custom directive) so that you can use the directives in a more conventional way and let sphinx do the formatting for you as a special case (decoupled from writing the documentation). But, as with LaTeX, it's probably best not to worry too much about the exact formatting, or at least to worry about the content first and the formatting second. Raw table syntax is always problematic in documentation because it is both hard to maintain the desired formatting consistently and it breaks up information that should be tightly programmatically coupled. I.e. once you've put useful information in three syntactically separate places (three table cells), neither Sphinx nor any other tool can use the relationship you intend between "RADICAL_BASE", "Root dir for temporary state files", and "`$HOME/.radical`".